io/dagreader.go

```diff
@@ -320,15 +320,10 @@ func (dr *dagReader) WriteTo(w io.Writer) (n int64, err error) {
 	})
 
 	if err == ipld.EndOfDag {
-		return n, io.EOF
-		// Reached the end of the (DAG) file, no more data to read.
-		// TODO: Is this a correct return error for `WriteTo`?
-	} else if err != nil {
-		return n, err
-		// Pass along any other errors from the `Visitor`.
+		return n, nil
```

Comment:
So I interpret that `EOF` shouldn't be returned in the `WriteTo` method, in that case I would still return any other error that's not `EndOfDag`, e.g.,

```
if err != nil && err != ipld.EndOfDag {
	return n, err
	// Pass along any other errors from the `Visitor` (except `EndOfDag`
	// which is expected).
} else {
	return n, nil
}
```

Reply:
It should return other errors - note `return n, **err**` below